### PR TITLE
guix: document when certain patches can be dropped

### PR DESCRIPTION
**Pull request by fanquake**
Additional notes for when patches can be dropped.

### CODE REVIEW

File: contrib/guix/patches/glibc-2.27-riscv64-Use-__has_include-to-include-asm-syscalls.h.patch [hunk 1]
```diff
@@ -4,6 +4,8 @@ See also: http://lists.busybox.net/pipermail/buildroot/2020-July/590376.html.
 
 https://sourceware.org/git/?p=glibc.git;a=commit;h=0b9c84906f653978fb8768c7ebd0ee14a47e662e
 
+This patch can be dropped when we are building with glibc 2.28+.
```
Comment (hebasto) [inline contrib/guix/patches/glibc-2.27-riscv64-Use-__has_include-to-include-asm-syscalls.h.patch:7]: https://sourceware.org/bugzilla/show_bug.cgi?id=24022:
> Target Milestone: | 2.29

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: contrib/guix/patches/glibc-2.27-fcommon.patch [hunk 2]
```diff
@@ -17,6 +17,8 @@ Date:   Fri May 6 11:03:04 2022 +0100
     https://sourceware.org/git/?p=glibc.git;a=commit;h=18363b4f010da9ba459b13310b113ac0647c2fcc
     https://sourceware.org/git/?p=glibc.git;a=commit;h=7650321ce037302bfc2f026aa19e0213b8d02fe6
 
+    This patch can be dropped when we are building with glibc 2.31+.
```
[the remark this answers is not in the document]
Comment (hebasto) [reply contrib/guix/patches/glibc-2.27-fcommon.patch:20]: If "2.31+" means ">=2.31", right?

[the remark this answers is not in the document]
Comment (fanquake) [reply contrib/guix/patches/glibc-2.27-fcommon.patch:20]: > According to https://sourceware.org/bugzilla/show_bug.cgi?id=26332:

According to the same thread it was backported to 2.31. Which is all that matters for us.

> "2.31+" means ">=2.31", right?

Yes.

File: contrib/guix/patches/glibc-2.27-riscv64-Use-__has_include-to-include-asm-syscalls.h.patch (hunk 1 above)
Comment (fanquake) [reply contrib/guix/patches/glibc-2.27-riscv64-Use-__has_include-to-include-asm-syscalls.h.patch:7]: > https://sourceware.org/bugzilla/show_bug.cgi?id=24022:

Again. The thread says it was backported to 2.28, which is what matters for us.

**Review by hebasto — APPROVED** [review]
ACK a09269a146b1e32a0e7979692f4455cc2f6faeae, I have reviewed the changes and they look OK.

**Review by jarolrod — COMMENTED** [review]
ACK a09269a146b1e32a0e7979692f4455cc2f6faeae

Can confirm the note on the long jump patch ;)

**Pull request merged**